File: src/main/java/xyz/xpto/gerenciamento/domain/entities/Projeto.java
package xyz.xpto.gerenciamento.domain.entities;

import java.time.LocalDate;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "projeto")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Projeto {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY, generator = "projeto_id_sequence")
    @SequenceGenerator(name = "projeto_id_sequence", sequenceName = "projeto_id_sequence", allocationSize = 1)
    @Column(name = "id")
    private long id;

    @Column(name = "nome", nullable = false)
    private String nome;

    @Column(name = "descricao", nullable = false)
    private String descricao;

    @Column(name = "data_inicio", nullable = false)
    private LocalDate dataInicio;

    @Column(name = "data_termino", nullable = false)
    private LocalDate dataTermino;

    @ManyToOne
    @JoinColumn(name = "equipe_id", nullable = false)
    private Equipe equipe;

    @ManyToOne
    @JoinColumn(name = "status_projeto_id", nullable = false)
    private StatusProjeto statusProjeto;

}

File: frontend/types/Projeto.ts
import type Equipe from "./Equipe"

type ProjetoParams = {
	id?: number
	nome?: string
	descricao?: string
	dataInicio?: string
	dataFim?: string
	equipe?: Equipe
	status?: string
}

export default class Projeto {
	public id: number;
	public nome: string;
	public descricao: string;
	public dataInicio: string;
	public dataFim: string;
	public equipe: Equipe | null;
	public status: string;

	constructor({ id, nome, descricao, dataInicio, dataFim, equipe, status }: ProjetoParams
	) {
		this.id = id || 0
		this.nome = nome || 'Sem nome'
		this.descricao = descricao || 'Sem descrição'
		this.dataInicio = dataInicio || ''
		this.dataFim = dataFim || ''
		this.equipe = equipe || null
		this.status = status || 'Sem status'
	}
}